### PR TITLE
Build system improvements

### PR DESCRIPTION
**Pull request by sethrj**
- Fail when `VECGEOM_CUDA_ARCHITECTURES` is specified on older versions of CMake (see #123 )
- Update external googletest submodule to eliminate warning under CMake 3.19 (@mrguilima )

### CODE REVIEW

**Review by mrguilima — COMMENTED** [review]
Thanks, do I need to do some kind of "module update" in my local area?

**Review by mrguilima — APPROVED** [review]

**Comment by sethrj** [comment]
> Thanks, do I need to do some kind of "module update" in my local area?

I think rerunning cmake will do it automatically, or you can run `git submodule update external/googletest`.  I'm going to wait for @pcanal 's response on #119 (or yours, since you had the same configuration and were the first one to encounter the kernel error) before merging though -- we may need to add additional error checking.

**Review by pcanal — APPROVED** [review]

**Pull request merged**
